python/TestHarness/RunParallel.py

```diff
@@ -148,6 +149,16 @@ def startReadyJobs(self, slot_check):
       os.chdir(saved_dir)
       sys.path.pop()
 
+  def getOutputFromFiles(self, tester):
+    # TODO: Use redirect_stdout when ready (libMesh update required Ref: https://github.com/libMesh/libmesh/issues/1159)
+    # file_list = glob.glob(os.path.join(tester.specs['test_dir'], tester.specs['redirect_stdout'] + '.*'))
+    file_list = glob.glob(os.path.join(tester.specs['test_dir'], 'stdout.processor.*'))
```

Comment:
I'd rather not glob, here's why, and thanks for bringing this up because we have a little more work to do on this PR.

Let's pick a `RunException` test **foo** and run it with the normal parallel argument to `./run_tests` (`-p`). That test (after the next libMesh update) will dump the following files: `foo_out.1 foo_out.2`. If we glob those, no problem. Now, imagine that we run the same test again but this time serial. It'll dump `foo_out.1`. Even if we delete these files before we run each test (the thingy we're currently missing), foo_out.2 will still be sitting on the file system so you'll accidentally pick it up in the glob!

Your first response might be, "let's just delete <filename>.* before we run the test but that's a TERRIBLE idea! We really don't want to run wildcard commands on the user machine in general. 

The better design is that we need to make sure that we only delete the files that we think we are going to generate. This means that for the redirection, that decision is going to have to be made by the TestHarness, not MOOSE. I'd suggest that we name it after the `test_name` since we already have that and it should be unique for a given directory. Depending on the number of CPUs (which we also know) we'll be able to not only delete the files that will be generated, but we'll be able to know what files we need to read too.